openedx/core/djangolib/testing/utils.py

```diff
@@ -190,3 +190,23 @@ def skip_unless_lms(func):
     Only run the decorated test in the LMS test suite
     """
     return skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in LMS')(func)
+
+
+def toggle_switch(name, active=True):
+    """
+    Activate or deactivate a Waffle switch. The switch is created if it does not exist.
+
+    Arguments:
+        name (str): name of the switch to be toggled.
+
+    Keyword Arguments:
+        active (bool): Whether the switch should be on or off.
+
+    Returns:
+        Switch
+    """
+    switch, __ = Switch.objects.get_or_create(name=name, defaults={'active': active})
+    switch.active = active
+    switch.save()
```

Comment:
Suggest using the 'create'  flag (from get_or_create) and if not created, check to see if there is a need to save to avoid unnecessary db activity